### PR TITLE
12183 fix special price

### PR DESCRIPTION
**Pull request by krnshah**
### Description (*)

Resolve the special price issue for configurable product.

### Fixed Issues (if relevant)

1. magento/magento2#12183: Configurable Product's normal price is not visible on Product Catalog

### Manual testing scenarios (*)

1. N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @krnshah. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-3994](https://jira.corp.magento.com/browse/ENGCOM-3994) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@krnshah thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by sivaschenko** [comment]
A performance test is required for these changes:
Category page with 50% configurable product with 1k option and 50% of bundle product with 1k option.
It can be done on a local environment with measurement of response time (just make it at least 10times) by any tools include a browser or apache benchmark.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-3994](https://jira.corp.magento.com/browse/ENGCOM-3994) has been created to process this Pull Request

**Comment by p-bystritsky** [comment]
@sivaschenko 
Seems like minor (7.5%) degradation is present on non-cached category pages. No degradation for cached.
Setup: category with 9 Configurable products with 1000 options each (generated with performance toolkit). 
2.3 mainline branch: 1.21 minutes average page load time (10 tries).
PR branch: 1.30 minutes average page load time (10 tries).

File: dev/tests/functional/tests/app/Magento/ConfigurableProduct/Test/Constraint/AssertConfigurableProductInCategory.php [hunk 1]
```diff
@@ -28,12 +27,19 @@ protected function assertPrice(FixtureInterface $product, CatalogCategoryView $c
...
-            number_format($price, 2, '.', ''),
-            $priceBlock->isOldPriceVisible() ? $priceBlock->getOldPrice() : $priceBlock->getPrice(),
-            'Product regular price on category page is not correct.'
-        );
-
+        if ($priceBlock->isNormalPriceVisible()) {
+            \PHPUnit\Framework\Assert::assertEquals(
+                number_format($price, 2, '.', ''),
+                $priceBlock->getNormalPrice(),
+                'Product normal price on category page is not correct.'
```
Comment (orlangur) [inline dev/tests/functional/tests/app/Magento/ConfigurableProduct/Test/Constraint/AssertConfigurableProductInCategory.php:34]: @p-bystritsky how is normal price different from regular price?

Comment (p-bystritsky) [reply dev/tests/functional/tests/app/Magento/ConfigurableProduct/Test/Constraint/AssertConfigurableProductInCategory.php:34]: @orlangur actual configurable product price on category page is located in span with 'normal_price' class.

**Comment by sivaschenko** [comment]
Hi @krnshah ! Unfortunately, we were unable to approve the functionality and the performance implications introduced by this pull request at this time and I have to close it.

**Pull request closed without merging**